src/pages/plugins/tutorials/plugin-lifecycle-hooks/index.md

```diff
@@ -0,0 +1,29 @@
+# Plugin Lifecycle events
+You can detect whether or not a plugin panel is visible with the help of UXP APIs
+
+<InlineAlert variant="info" slots="text"/>
+
+This is only applicable to plugins with a panel. You cannot detect the execution of a command plugin via these hooks. //TODO verify fact about command plugins and hooks. Also, for plugins that have multiple panels, you cannot differentiate between them, at present. Familiarize yourself with [plugin entrypoints](../../concepts/entry-points/) to know the difference between a command and panel plugin.
+
+
+## Example
+
+<CodeBlock slots="heading, code" repeat="1" languages="JavaScript" />
+
+#### JavaScript
+```js
+document.addEventListener('uxpcommand', (event) => {
+    const { commandId } = event;
+    if (commandId === 'uxpshowpanel') {
+      console.log("Showing plugin panel");
+    } else if (commandId === 'uxphidepanel') {
+      console.log('Hiding plugin panel');
+    }
+});
+```
+
+<!-- 
+## Reference material
```

Comment:
@kareemmyk  Missing doc. UXP API reference doesnt capture these APIs. Please add to backlog.

Reply:
backlog created.
https://jira.corp.adobe.com/browse/UXP-21657

Reply:
@anoopvalomkot  I wasnt able to make these work within InDesign. Are these supported?